analysis/vertfn/analyzer.go

```diff
@@ -25,34 +27,53 @@ var (
 	colorize bool
 )
 
+type RefKind string
+
+const (
+	Func     RefKind = "func"
+	Type     RefKind = "type"
+	RecvType RefKind = "recvtype"
+	Value    RefKind = "value"
+)
+
+type Direction string
+
+const (
+	Down   Direction = "down"
+	Up     Direction = "up"
+	Ignore Direction = "ignore"
+)
+
+var RefOrder = map[RefKind]Direction{
+	Func:     Down,
+	Type:     Down,
+	RecvType: Up,
+	Value:    Down,
+}
+
 func init() {
 	Analyzer.Flags.BoolVar(&verbose, "verbose", false, `print all details`)
 	Analyzer.Flags.BoolVar(&colorize, "color", true, `colorize terminal`)
-}
-
-func fnameFromFuncDecl(n *ast.FuncDecl) string { return n.Name.Name }
-
-func fnameFromCallExpr(n *ast.CallExpr) []string {
-	if ind, ok := n.Fun.(*ast.Ident); ok && ind != nil {
-		return []string{ind.Name}
-	}
-	if sel, ok := n.Fun.(*ast.SelectorExpr); ok && sel != nil {
-		var fnames []string
-		fnames = append(fnames, sel.Sel.Name)
-		if call, ok := sel.X.(*ast.CallExpr); ok && call != nil {
-			fnames = append(fnames, fnameFromCallExpr(call)...)
-		}
-		return fnames
+	addDirectionFlag := func(kind RefKind, desc string) {
+		Analyzer.Flags.Func(string(kind)+"-dir", fmt.Sprintf("%s (default %s)", desc, RefOrder[kind]), func(s string) error {
```

Comment:
instead of direct string cast, better String method.
```suggestion
		Analyzer.Flags.Func(kind.String()+"-dir", fmt.Sprintf("%s (default %s)", desc, RefOrder[kind]), func(s string) error {
```

Reply:
why is a String method better?

Reply:
if underlying type is not string anymore, but say struct, then direct string cast will not compile anymore. Generally way to deal with this is to define String() method. Then any type can define it and this code will still compile.